File: Data-Structures/Arrays/customArray.py
class CustomArray:
    def __init__(self, capacity):
        self.data = [0] * capacity
        self.size = 0
    
    def add(self, value):
        if self.size == len(self.data):
            self.resize()
        self.data[self.size] = value
        self.size += 1
    
    def resize(self):
        new_capacity = len(self.data) * 2
        new_data = [0] * new_capacity

        for i in range(self.size):
            new_data[i] = self.data[i]

        self.data = new_data

    def get(self, index):
        if index < 0 or index >= self.size:
            raise IndexError("Index is out of bounds")
        return self.data[index]
    
    def removeAt(self, index):
        if index < 0 or index >= self.size:
            raise IndexError("Index is out of bounds")

        
        for i in range(index, self.size - 1):
            self.data[i] = self.data[i + 1]

        self.size -= 1   

    def display(self):
        for i in range(self.size):
            print(self.data[i])

    def get_size(self):
        return self.size
    

if __name__ == "__main__":
    array = CustomArray(12)
    array.add(23)
    array.add(34)
    array.add(56)
    array.add(21)
    array.add(29)
    array.add(87)
    array.add(43)


    print("Initial size:", array.get_size())
    array.removeAt(6)
    print("\nArray after removing index 6:")
    array.display()
    print("\nFinal size:", array.get_size())

File: Data-Structures/Arrays/customArray.java

public class customArray{

    private int [] data;
    private int size;

    public customArray(int capacity){
        data = new int[capacity];
        size = 0;
    }


    public void add(int value){
        if(size == data.length){
            resize();
        }
        data[size] = value;
        size++;
    }

    private void resize(){
        int[] newData = new int[data.length * 2];
        for(int i = 0; i < data.length; i++){
            newData[i] = data[i];
        }
        data = newData;
    }

    public int get(int index){
        if(index < 0 || index >= size){
            throw new IndexOutOfBoundsException("Index is out of bound");
        }
        return data[index];
    }

    public void print(){
        for(int i = 0; i < size; i++){
            System.out.println(data[i]);
        }
    }

    public int size(){
        return size;
    }

    public static void main(String[] args){
        customArray array1 = new customArray(7);

       
        array1.add(7);
        array1.add(23);
        array1.add(21);
        array1.add(45);
        array1.add(56);
        array1.add(87);
        array1.add(76);
        array1.add(743);
        

        array1.print();
        System.out.println("The size of an array is:" + array1.size());
        System.out.println( array1.get(3));
    }
}
